### DISCUSSION TITLE
How to Add custom button?

### DISCUSSION THREAD

**Discussion by axelcasauran** [serenity-is/Serenity#6261] [Help] [answered]
Hi, 

I'm new here. 
I would like to ask how to add custom button in the **grid** (next to New button)?
If there is a documentation already, please reply with the link.

Thank you.

**Comment by edwardch** [comment, marked as answer]
Hey Axel,

Check out the common-features repo for some examples. 

[ProductGrid.ts](https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Serenity.Demo.Northwind/Product/ProductGrid.ts) has some examples on adding buttons.

buttons.push is what you want, and in this example look at the "Save Changes" button. You'll want to put your own method into the onclick option...

Hope that helps!

**Reply by axelcasauran** [reply]
I tried adding my own extension "typings\serenity.extensions\index.d.ts" (_Serenity.Extensions_) but it seems like it overwritten every time I run my project; my codes disappears. Is this only available for Paid/Premium packages? or am I missing something?

**Reply by edwardch** [reply]
Most (all?) of the files in the typings directory auto generated from the TypeScript files in your Modules.

You want to edit 
`YourModule/YourClass/YourClassGrid.ts`

You can also modify this file while your project is running and it should recompile the js so you won't need to restart your project after each change.

**Reply by axelcasauran** [reply]
awesome!! works like a charm! thank you!
